File: base_code/app/src/main/java/com/dehaat/dehaatassignment/dao/BookDao.java
package com.dehaat.dehaatassignment.dao;

import com.dehaat.dehaatassignment.model.Book;
import com.dehaat.dehaatassignment.model.books;

import java.util.ArrayList;
import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface BookDao {

    @Insert
    void insertAll(Book book);

    @Query("select * from Book where author_name = :authorName")
    List<Book> getBooks(String authorName);


}

File: base_code/app/src/main/java/com/dehaat/dehaatassignment/dao/BookDao.kt
package com.dehaat.dehaatassignment.dao

import afu.org.checkerframework.checker.igj.qual.Mutable
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.dehaat.dehaatassignment.model.Book
import com.dehaat.dehaatassignment.model.books

import java.util.ArrayList

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface BookDao {

    @Insert
    suspend fun insertAll(book: Book)

    @Query("select * from Book where author_name = :authorName")
     fun getBooks(authorName: String?): LiveData<Book>


}
